Brokerages/SymbolPropertiesDatabaseSymbolMapper.cs

```diff
@@ -0,0 +1,182 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using QuantConnect.Securities;
+
+namespace QuantConnect.Brokerages
+{
+    /// <summary>
+    /// Provides the mapping between Lean symbols and brokerage symbols using the symbol properties database
+    /// </summary>
+    public class SymbolPropertiesDatabaseSymbolMapper : ISymbolMapper
+    {
+        private readonly string _market;
+
+        // map Lean symbols to symbol properties
+        private readonly Dictionary<SecurityDatabaseKey, SymbolProperties> _symbolPropertiesMap;
```

Comment:
Based on the usages, input is always Symbol, maybe the `Key` here should directly be a `Symbol` or even better a `SecurityIdentifier`?